src/NuGet.Clients/NuGet.VisualStudio.Common/ServiceLocator.cs

```diff
@@ -97,8 +93,8 @@ public static void InitializePackageServiceProvider(IServiceProvider provider)
 
```

Comment:
Is this `SwitchToMainThreadAsync()` call necessary?  (Reread the issue and talk to Ryan if necessary.)

Reply:
It's not. 
Fixed.

Reply:
Actually now I'm finding different resources that suggest otherwise.

Reply:
The cast at the line below might involve com operation and since this is generic, we can't predict whether it does. 

I'll ping Ryan to confirm.

Reply:
Looking at my PR further I think I might need to use the free threaded utility that you added in your PR. 

https://github.com/NuGet/NuGet.Client/pull/2889/files#diff-6b14c4c96925a5094908e8cb833d59b4R102

And then GetComponentModelAsync with that rather than GetGlobalServiceAsync.

Reply:
I have updated my PR & reached out to  Ryan. 

Will work with him further.

Reply:
A basic summary of his reply: 

my basic argument is that you are ‘async’ service retrieval just switched to the UI thread and then call GetService, which is synchronous. If that call forces a package load it now occurs on the UI thread. Alternatively, if you switched to the UI thread and called await GetServiceAsync, if that service was async, or came from a package that was async, it could move the dll load off the UI thread which is a big perf win on slow harddisk machines (or machines with fast harddisks that are just busy with other things, like running a complex build).

The one caveat is that because a managed cast of an STA (native) service in VS triggers RPC if done off the UI thread, you still generally need to be on the UI thread before you call GetServiceAsync. 

So I guess the end result is your GetServiceAsync wrapper should

1: Switch to the main thread using SwitchToMainThreadAsync (I think it already does this)
2: Always use the async service provider once its on the UI thread (await GetServiceAsync)

#2 gets your perf wins, it would be ideal if you could avoid #1 as the service provider itself will switch if it really needs to, but unfortunately since we don’t know the final interface you may cast the object to the cast would still occur on the calling (b/g) thread, which would trigger RPC. So to avoid this we just tell people to get to the UI thread before they call GetServiceAsync.